### ISSUE TITLE
[bug] File and folder name collisions in `copy()`.

### ISSUE THREAD

**Issue by nmccrea** [conan-io/conan#13675]
### Environment details

* Operating System+version: macOS 13.2.1
* Conan version: 1.59
* Python version: 3.10.5

### Steps to reproduce

I am trying to package a file and a folder together that happen to have the same name, although with different case. We are working exclusively on case-sensitive file systems, in which these two items can definitely coexist in the same folder, so I'm hoping I can get this working without having to rename things.

Example: My `package()` function looks something like this:

```python
# There is a subdirectory, `folders/FOO/`, which will be copied
self.copy("*", src="folders", dst="bin")

# `foo` is a binary executable
self.copy("foo", src="build/Release", dst="bin")
```

I would like to get the following structure in the package folder:

```
- bin/
  | - foo
  | - FOO/
      | - ...
```

Instead I get the following strange results:

```
- bin/
  | - FOO
      | - foo
      | - ...
```

In other words, instead of copying `foo` to `bin/foo`, it is for some reason placed in `bin/FOO/foo`.

Passing `ignore_case=False` has no effect.

I also tried reversing the order of the copy statements, which results in a `NotADirectoryError` when copything the folder:

```
ERROR: package/x.y.z: Error in package() method, line 105
        self.copy("*", src="folders", dst="bin")
        NotADirectoryError: [Errno 20] Not a directory: '.../bin/FOO/some_file_name'
```

It is looking like I will have to rename either the file or the folder, which I was hoping to avoid since either will have some knock-on effects. Before I do that I wanted to check if there is something I am missing or doing wrong, or if it is possible to fix the unexpected behavior.

Thanks.

**Comment by nmccrea** [comment]
I've renamed my files to resolve my issue. I'll close this so it doesn't clutter the repo. I do think this is still undesirable behavior but it is a less common case and the ideal solution generally is to avoid naming collisions in the first place.

**Comment by memsharded** [comment]
Hi @nmccrea 

Thanks for your report, sorry we didn't answer earlier.

Not sure what could be happening there. In any case, it is recommended to use the ``from conan.tools.files import copy`` function instead of ``self.copy()``. It uses absolute  paths as ``src`` and ``dst``, so it is possible that it has less issues regarding possible ambiguity with folders. I'd recommend using them, and if there is still an issue, report with the new ``copy()`` function, and we can have a look.
